containers/ReferenceValue.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { changeReferenceValue } from '../actions'
+
+let ReferenceValue = ({ entityId, refId, referenceValue, dispatch }) => (
+  <div>
+    <input type="text" 
+      placeholder="reference value"
+      value={referenceValue}
+      onChange={e => dispatch(changeReferenceValue(entityId, refId, e.target.value))}/>
```

Comment:
And for onChange, I suggest not to define action for every one. They use actually same state.
Just define one in EntityPanel (root)
and children all pass their onchange event to parent's. collect all to root and then call one action